spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractConsumerSeekAware.java

```diff
@@ -82,20 +86,20 @@ public void unregisterSeekCallback() {
 	}
 
 	/**
-	 * Return the callback for the specified topic/partition.
+	 * Return the callbacks for the specified topic/partition.
 	 * @param topicPartition the topic/partition.
-	 * @return the callback (or null if there is no assignment).
+	 * @return the callbacks (or null if there is no assignment).
 	 */
 	@Nullable
-	protected ConsumerSeekCallback getSeekCallbackFor(TopicPartition topicPartition) {
+	protected List<ConsumerSeekCallback> getSeekCallbacksFor(TopicPartition topicPartition) {
 		return this.callbacks.get(topicPartition);
 	}
 
 	/**
 	 * The map of callbacks for all currently assigned partitions.
 	 * @return the map.
 	 */
-	protected Map<TopicPartition, ConsumerSeekCallback> getSeekCallbacks() {
+	protected Map<TopicPartition, List<ConsumerSeekCallback>> getSeekCallbacks() {
```

Comment:
It should be mentioned in the `whats-new` section since this is a breaking change.

Reply:
This one is a bit harder.
May be let's call a new method as `getAllSeekCallbacks()`?
With a `TODO` that it is supposed to be deprecated and removed in the future when we remove the old one?

Reply:
`getAllSeekCallbacks()` is good. however, I don't understand a bit the below comment.
> With a TODO that it is supposed to be deprecated and removed in the future when we remove the old one?

Do you mean to add `getAllSeekCallbacks()` first, and then deprecate it later after modifying the `getSeekCallbacks()` method internals?

Reply:
Sorry for not being Cristal clear. So, the plan is:
1. Deprecate existing method
2. Introduce new one with that “All” awkward name.
3. Remove that deprecated method in the next release.
4. Next release Introduce a new method with the desired contract and the name of removed before.
5. Deprecate that “All” awkward method.
6. And then remove it eventually .

This way we have a smooth migration path without breaking changes for target projects .

However I feel like with this releases cycle we would be far in the future where at the same time we may already start a major 4.0 version. And that’s where we can skip some of the backward compatibility steps and have relief with freedom to introduce a breaking change 🤪

Reply:
Thanks for explanation.

Actually, Instead of using the old `callbacks` field, I added the `topicToCallbacks` field. 
> [3570f7a](https://github.com/spring-projects/spring-kafka/pull/3341/commits/3570f7a53a6fc9102302b5b80bb1e7610695017c)

This way, not only will existing features work the same, but it will also naturally accommodate new features.

However, this seems to make it more complicated, so please feel free to take a look and let me know what you think. 

Feel free to comment on the javaDocs as well.
The `whats-new` section documentation will be written after the code modifications are complete.

Reply:
To summarize, I like your strategy above, but maybe we could make a clearer distinction between `deprecated` and `new` methods. To that end, I did something like this.

- **To be deprecated**
  - `callbacks` field
  - `getSeekCallbackFor()`
  - `getSeekCallbacks()`

- **To be introduced**
  - `topicToCallbacks` field
  - `getSeekCallbacksFor()`
  - `getTopicsAndCallbacks()`